src/ethereum/prague/fork.py

```diff
@@ -1023,29 +1023,26 @@ def process_transaction(
 
     output = process_message_call(message, env)
 
-    # For EIP-7623 gas_used includes the intrinsic costs with the floor price
-    # for calldata.
-    gas_used = tx.gas - output.gas_left
+    # For EIP-7623 we first calculate the execution_gas_used, which includes
+    # the execution gas refund.
+    execution_gas_used = tx.gas - output.gas_left
+    gas_refund = min(execution_gas_used // 5, output.refund_counter)
+    execution_gas_used -= gas_refund
 
     # EIP-7623 floor price (note: no EVM costs)
-
-    # Transactions with less gas used than the floor pay at the floor cost.
     floor = Uint(tokens_in_calldata * FLOOR_CALLDATA_COST + TX_BASE_COST)
```

Comment:
I don't like the `floor` name. The EIP don't really defines this name like this except for one constant name. How about `min_gas_cost` or `floor_gas_cost`?